deploy/helm/sumologic/templates/metrics-configmap.yaml

```diff
@@ -1,3 +1,4 @@
+{{- if eq .Values.fluentd.metrics.enabled true }}
```

Comment:
I used this flag as this is part of `fluentd` itself. and we already have `fluentd.events.enabled` as compared to traces which is not part of the fluentd section.

Also I feel having both `sumologic.metrics.enabled` and `fluentd.metrics.enabled` will not add any functionality as it can be covered by only one.

Thoughts? cc @frankreno @samjsong

Reply:
hmm, I'm not sure I understand the usecase for having both `sumologic.metrics.enabled` and `fluentd.metrics.enabled`. I can't think of a case where you would want the fluentd pipeline enabled but not the metrics resources, or vice versa. @perk-sumo can you share your reasoning please?

Reply:
I feel we should have `sumologic.metrics.enabled` instead of `fluentd.metrics.enabled`, especially if we gonna replace fluentd in the future, but it's meaningful change, so I would stick to the current flag

Reply:
The idea is to have global flag that is not attached to the underlying implementation so that we (or the user) can switch the tool easily.
There are couple of usecases:
* disable all the metrics, all the fluentd's, prometheus, configmaps etc - set `sumologic.metrics.enabled=false`
* enable all the metrics in the default mode: `sumologic.metrics.enabled=true`
* enable metrics, but use own prometheus: `sumologic.metrics.enabled=true` and `prometheus-operator.enabled=false`
* enable metrics, but use own fluentd for tagging: `sumologic.metrics.enabled=true` and `fluentd.metrics.enabled=false`
* enable metrics, but use experimental OTC for that: `sumologic.metrics.enabled` and `fluentd.metrics.enabled=false`, `prometheus-operator.enabled=false`, `otelcol.metrics.enabled=true`

Reply:
What @perk-sumo suggestion seems to be the cleanest approach.